### PR TITLE
Remove services endpoint and migrate functionality to CLI

### PR DESCRIPTION
**Pull request by JoeMcGuire98**
### Proposed changes
We have an opportunity to completely remove our API server. Part of that includes migrating the /services endpoint, which returns information about the services that are a part of the mesh. We could instead just have our CLI get this information through the k8s API, and format the output.

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-service-mesh/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto main
- [x] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: api/mesh.yaml [hunk 1]
```diff
@@ -218,73 +218,6 @@ paths:
...
-                    ],
-                    "serviceIP": "10.40.3.184"
-                  }
-                ]
-        "401":
-          $ref: '#/components/responses/Unauthorized'
-        "403":
-          $ref: '#/components/responses/Forbidden'
         "500":
           $ref: '#/components/responses/InternalServerError'
```
Comment (sjberman) [inline api/mesh.yaml:222]: Can delete this 500

File: internal/nginx-meshctl/commands/services.go [hunk 2]
```diff
@@ -16,10 +20,20 @@ const longServices = `List the Services registered with NGINX Service Mesh.
 - Outputs the Services and their upstream addresses and ports.
 - The list contains only those Services whose Pods contain the NGINX Service Mesh sidecar.
 `
+const injectionLabel = "injector.nsm.nginx.com/auto-inject"
+
+var IgnoredNamespaces = map[string]bool{
+	"kube-system": true,
+}
```
Comment (sjberman) [inline internal/nginx-meshctl/commands/services.go:27]: These both should already be constants in the `mesh` package.

File: internal/nginx-meshctl/commands/services.go [hunk 3]
```diff
@@ -96,3 +110,49 @@ func GetServices() *cobra.Command {
...
+	nsObj := &v1.Namespace{}
+	if err := k8sClient.Get(ctx, client.ObjectKey{
+		Namespace: "",
+		Name:      ns,
+	}, nsObj); err != nil {
+		fmt.Printf("error getting namespace: %v\n", err)
+		return false, err
+	}
+	return slices.Contains(*meshConfig.EnabledNamespaces, ns) ||
+		nsObj.GetLabels()[injectionLabel] == "enabled" ||
```
Comment (sjberman) [inline internal/nginx-meshctl/commands/services.go:156]: "enabled" is also a constant that we have defined.

Comment (JoeMcGuire98) [reply internal/nginx-meshctl/commands/services.go]: Didn't see this defined as a constant anywhere in the `mesh` package, so I added a constant definition for it and used that

Comment (sjberman) [reply internal/nginx-meshctl/commands/services.go]: Oh you're right, that's actually in our control plane repo.

File: api/mesh.yaml [hunk 4]
```diff
@@ -218,8 +218,6 @@ paths:
           $ref: '#/components/responses/Unauthorized'
         "403":
           $ref: '#/components/responses/Forbidden'
-        "500":
-          $ref: '#/components/responses/InternalServerError'
```
Comment (sjberman) [inline api/mesh.yaml:289]: Be sure to regenerate after making this change.

**Review by sjberman — APPROVED** [review]

File: internal/nginx-meshctl/commands/services.go [hunk 5]
```diff
@@ -29,60 +38,60 @@ func GetServices() *cobra.Command {
...
-			return mesh.ParseAPIError(res)
-		}
 
-		data, err := mesh.ParseGetServicesResponse(res)
-		if err != nil {
-			fmt.Println(genericGetServicesErrorInfo)
-
-			return fmt.Errorf("error parsing response body from NGINX Service Mesh API Server: %w", err)
+		for _, serviceObj := range services.Items {
+			serviceObj := serviceObj
```
Comment (chase-kiefer) [inline internal/nginx-meshctl/commands/services.go:57]: This can probably be deleted

File: internal/nginx-meshctl/commands/services.go [hunk 6]
```diff
@@ -29,60 +38,60 @@ func GetServices() *cobra.Command {
...
+				if err != nil {
+					return epErr
+				}
+				newDetails := serviceDetails{
+					name:      serviceObj.Name,
+					namespace: serviceObj.Namespace,
+					ports:     serviceObj.Spec.Ports,
+					addresses: upstreams,
+				}
+				meshServices = append(meshServices, newDetails)
```
Comment (chase-kiefer) [inline internal/nginx-meshctl/commands/services.go:70]: We can probably just create and append the new serviceDetails instance in one go.

File: internal/nginx-meshctl/commands/services.go [hunk 7]
```diff
@@ -96,3 +105,49 @@ func GetServices() *cobra.Command {
...
+// isNamespaceInjectionEnabled returns whether a given namespace has injection enabled.
+func isNamespaceInjectionEnabled(ctx context.Context, k8sClient client.Client, ns string) (bool, error) {
+	meshClient, err := mesh.NewMeshClient(initK8sClient.Config(), meshTimeout)
+	if err != nil {
+		return false, fmt.Errorf("failed to get mesh client: %w", err)
+	}
+	meshConfig, err := GetMeshConfig(meshClient)
+	if err != nil {
+		return false, fmt.Errorf("unable to get mesh config: %w", err)
+	}
```
Comment (chase-kiefer) [inline internal/nginx-meshctl/commands/services.go:141]: This could be worth pulling out into it's own function

Comment (sjberman) [reply internal/nginx-meshctl/commands/services.go:141]: FWIW, this is gonna go away with my API migration work and be replaced.

**Review by chase-kiefer — APPROVED** [review]
LGTM

**Pull request merged**